ToDoApp/ToDoApp.Services/Services/ToDoTasksServices.cs

```diff
@@ -0,0 +1,167 @@
+                .Where(x => x.AssigneeId == _currentUserServices.UserId && x.BoardId == boardId)
+                .Select(x => new GetTaskDto
+                {
+                    Id = x.Id,
+                    Title = x.Title,
+                    Description = x.Description,
+                    BoardId = x.BoardId,
+                    StatusId = x.StatusId,
+                    AssigneeId = x.AssigneeId
+                })
+                .ToListAsync();
+        }
+
+        public async Task CreateTaskAsync(int boardId, CreateTaskDto taskDto)
+        {
+            var board = await _context.Board.FindAsync(boardId);
+
+            if (board is null)
+            {
+                throw new BoardNotFoundException();
+            }
+
+            var item = new Tasks
+            {
+                Title = taskDto.Title,
+                Description = taskDto.Description,
+                BoardId = boardId,
+                StatusId = 1,
+                AssigneeId = _currentUserServices.UserId
+            };
+
+            await _context.Task.AddAsync(item);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task UpdateTaskTitleAsync(UpdateTaskDto updateTaskDto)
+        {
+            var task = await _context.Task.FindAsync(updateTaskDto.Id);
+
+            if (task is null) 
+            {
+                throw new TaskNotFoundException();
+            }
+
+            if (task.BoardId != updateTaskDto.BoardId)
+            {
+                throw new TaskHasDifferentBoardException();
+            }
+
+            task.Title = updateTaskDto.Title;
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task UpdateTaskDescriptionAsync(UpdateTaskDto updateTaskDto)
+        {
+            var task = await _context.Task.FindAsync(updateTaskDto.Id);
+
+            if (task is null)
+            {
+                throw new TaskNotFoundException();
+            }
+
+            if (task.BoardId != updateTaskDto.BoardId)
+            {
+                throw new TaskHasDifferentBoardException();
+            }
+
+            task.Description = updateTaskDto.Description;
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task UpdateTaskStatusAsync(UpdateTaskDto updateTaskDto)
+        {
+            var task = await _context.Task.FindAsync(updateTaskDto.Id);
+
+            if (task is null)
+            {
+                throw new TaskNotFoundException();
+            }
+
+            if (task.BoardId != updateTaskDto.BoardId)
+            {
+                throw new TaskHasDifferentBoardException();
+            }
+
+            var statusValidation = await _context.statusesValidations.Where(x => x.StatusId == task.StatusId && x.StatusValidationId == updateTaskDto.StatusId).FirstOrDefaultAsync();
+
+            if (statusValidation is null)
+            {
+                throw new InvalidTaskStatusException();
+            }
+            else
```

Comment:
the else block is redundant, the status assignment can be on top level and it will make it a bit easier to read